### DISCUSSION TITLE
doesn't work; can't find config file

### DISCUSSION THREAD

**Discussion by iamnewton** [textlint/textlint#1456] [Q&A]
**What version of textlint are you using?**

14.3.0

**What file type (Markdown, plain text, etc.) are you using?**

Markdown, but doesn't matter

**What did you do? Please include the actual source code causing the issue.**

1. installed `textlint` with some plugins
2. ran `textlint --init`
3. updated config with the following

```json
{
  "plugins": {},
  "filters": {
		"comments": true
  },
  "rules": {
    "common-misspellings": true,
    "en-spell": true,
    "no-dead-link": true,
    "no-empty-section": true,
    "preset-ja-technical-writing": true,
    "prh": true,
    "spellcheck-tech-word": true,
		"terminology": {
			"terms": [
				{ "term": "frontend", "suggestion": "front-end" }
			],
		},
  }
}
```

4. ran `./node_modules/.bin/textlint **/*.md --config .textlintrc.json`

**What did you expect to happen?**

It run successfully.

**What actually happened? Please include the actual, raw output from textlint.**

It barfed.

```
./node_modules/.bin/textlint **/*.md --config .textlintrc.json
(node:44561) [DEP0040] DeprecationWarning: The `punycode` module is deprecated. Please use a userland alternative instead.
(Use `node --trace-deprecation ...` to show where the warning was created)

== No rules found, textlint hasn’t done anything ==

Possible reasons:
* Your textlint config file has no rules.
* You have no config file and you aren’t passing rules via command line.
* Your textlint config has a syntax error.

=> How to set up rules?
https://github.com/textlint/textlint/blob/master/docs/configuring.md
```

`$ npx --no-install textlint --debug target.md` Output:

```
./node_modules/.bin/textlint --debug README.md
  textlint:cli cliOptions: {"debug":true,"ignorePath":"/code/.textlintignore","init":false,"stdin":false,"format":"stylish","color":true,"quiet":false,"textlintrc":true,"cache":false,"cacheLocation":"/code/.textlintcache","_":["README.md"]} +0ms
  textlint:cli textlint --version: 14.3.0 +1ms
  textlint:cli Running on files, stdin-filename: undefined +0ms
  textlint:cli cliDescriptor: {"rule":[],"filterRule":[],"plugin":[]} +0ms
  textlint:loader:TextlintrcLoader loadTextlintrc failed: { ok: false, error: { message: 'textlint config is not found', errors: [ ReferenceError: Cannot read config file: /code/.textlintrc.js Error: JavaScript is not defined at Object.<anonymous> (/code/.textlintrc.js:18:16)     at Module._compile (node:internal/modules/cjs/loader:1572:14) at requireFromString (/code/node_modules/require-from-string/index.js:28:4) at loadJSConfigFile (/code/node_modules/rc-config-loader/lib/rc-config-loader.js:130:50) at findConfig (/code/node_modules/rc-config-loader/lib/rc-config-loader.js:82:28) at rcFile (/code/node_modules/rc-config-loader/lib/rc-config-loader.js:52:12) at loadRawConfig (/code/node_modules/@textlint/config-loader/lib/src/config-loader.js:106:55) at loadConfig (/code/node_modules/@textlint/config-loader/lib/src/config-loader.js:71:55) at loadTextlintrc (/code/node_modules/textlint/lib/src/loader/TextlintrcLoader.js:36:57) at loadDescriptor (/code/node_modules/textlint/lib/src/cli.js:27:55) ] } } +0ms
  textlint:cli textlintrcDescriptor: {"rule":[],"filterRule":[],"plugin":[{"id":"@textlint/textlint-plugin-text","options":{}},{"id":"@textlint/textlint-plugin-markdown","options":{}}]} +1ms
  textlint:cli mergedDescriptor: {"rule":[],"filterRule":[],"plugin":[{"id":"@textlint/textlint-plugin-text","options":{}},{"id":"@textlint/textlint-plugin-markdown","options":{}}]} +0ms

== No rules found, textlint hasn’t done anything ==

Possible reasons:
* Your textlint config file has no rules.
* You have no config file and you aren’t passing rules via command line.
* Your textlint config has a syntax error.

=> How to set up rules?
https://github.com/textlint/textlint/blob/master/docs/configuring.md```

**Comment by azu** [comment]
> $ npx --no-install textlint --debug target.md Output:
>   textlint:loader:TextlintrcLoader loadTextlintrc failed: { ok: false, error: { message: 'textlint config is not found', errors: [ ReferenceError: Cannot read config file: /code/.textlintrc.js Error: JavaScript is not defined at Object.<anonymous> (/code/.textlintrc.js:18:16)     at Module._compile (node:internal/modules/cjs/loader:1572:14) at requireFromString (/code/node_modules/require-from-string/index.js:28:4) at loadJSConfigFile (/code/node_modules/rc-config-loader/lib/rc-config-loader.js:130:50) at findConfig (/code/node_modules/rc-config-loader/lib/rc-config-loader.js:82:28) at rcFile (/code/node_modules/rc-config-loader/lib/rc-config-loader.js:52:12) at loadRawConfig (/code/node_modules/@textlint/config-loader/lib/src/config-loader.js:106:55) at loadConfig (/code/node_modules/@textlint/config-loader/lib/src/config-loader.js:71:55) at loadTextlintrc (/code/node_modules/textlint/lib/src/loader/TextlintrcLoader.js:36:57) at loadDescriptor (/code/node_modules/textlint/lib/src/cli.js:27:55) ] } } +0ms

> ReferenceError: Cannot read config file: /code/.textlintrc.js Error: JavaScript is not defined

Is there a `.textlintrc.js` instead of `.textlintrc.json`?

If you use `.textlintrc.js`, you need to export a object 
```js
module.exports = { ... }
```

**Comment by reithose** [comment]
Did you install all the rules you added to your config file? If you configured a rule, but didn't install it, you get the same error.
